src/Microsoft.Extensions.CommandLineUtils/CommandLine/CommandLineApplication.cs

```diff
@@ -146,10 +147,17 @@ public int Execute(params string[] args)
                     if (longOption != null)
                     {
                         processed = true;
-                        option = command.GetOptions().SingleOrDefault(opt => string.Equals(opt.LongName, longOption[0], StringComparison.Ordinal));
+                        var longOptionName = longOption[0];
+                        option = command.GetOptions().SingleOrDefault(opt => string.Equals(opt.LongName, longOptionName, StringComparison.Ordinal));
 
                         if (option == null)
                         {
+                            if (string.IsNullOrEmpty(longOptionName) && !command._throwOnUnexpectedArg  && AllowArgumentSeparator)
```

Comment:
It puts everything after the first '--' in app.RemainingArguments.

Example: if specified twice, `dotnet command -- --`, then `app.RemainingArguments == new [] { "--" }`